src/main/scala/zio/prelude/Validation.scala

```diff
@@ -183,9 +184,9 @@ object Validation extends LowPriorityValidationImplicits {
    * Derives an `Equal[Validation[E, A]]` given an `Equal[E]` and an
    * `Equal[A]`.
    */
-  implicit def ValidationEqual[E, A: Equal]: Equal[Validation[E, A]] =
+  implicit def ValidationEqual[E: Equal, A: Equal]: Equal[Validation[E, A]] =
     Equal.make {
-      case (Failure(es), Failure(e1s)) => es.groupBy(identity) == e1s.groupBy(identity)
+      case (Failure(es), Failure(e1s)) => es.groupBy(identity) === e1s.groupBy(identity)
```

Comment:
We aren't actually able to use a custom definition of equality for `E` here because we are comparing two `Map` values that use Scala's default notion of equality and hash code for the keys. So let's not pretend otherwise.

Reply:
Sure, that's what's currently going to take place, but isn't it better to delegate this behaviour to `Equal[Map[A, B]] where B: Equal` anyway?
The type of `es.groupBy(identity)` is `MapEqual[E, Chunk[E]]`, so the `Equal` instance is used for `Chunk[E]`.
Maybe that's something worth it? It would at least be consistent (in the sense that we're using `===` in the definitions of `Equal` as much as possible).

Let me know, if you would prefer `==` here, no big deal, I'm just trying to figure out what's better by thinking out loud.

Reply:
:heavy_check_mark: